abrechnung/sftpgws.py

```diff
@@ -91,17 +82,34 @@ async def run(self):
                                            print=None)
             finally:
                 # deregister at db
-                await conn.execute("SELECT forwarder_stop($1);",
+                await conn.execute("select * from forwarder_stop($1);",
                                    self.cfg['websocket']['id'])
 
-    def psql_message(self, connection, pid, channel, payload):
+    def psql_notification(self, connection, pid, channel, payload):
         """
         this is called by psql to handle a notify.
         """
+        del connection, pid  # unused
+
+        payload_json = json.loads(payload)
+        message = json.dumps({
+            "type": "notification",
+            "event": payload_json["event"],
+            "args": payload_json["args"]
+            # TODO: support for connection_id-specific 'triggers' and 'count'
+        }) + "\n"
+
+        if payload_json['connections'] == '*':
+            tx_queues = self.tx_queues.values()
+        else:
+            tx_queues = (
+                self.tx_queues[connection_id]
+                for connection_id in payload_json['connections']
+                if connection_id in self.tx_queues
+            )
```

Comment:
better use a real loop and fill `tx_queues` and also check if the queue is actually known (else skip it)

Reply:
queue is known?

Reply:
ah right sli